flask_helpers.py

```diff
@@ -1,33 +1,108 @@
 import os
+import re
 from functools import wraps
 
 from flask import redirect, render_template, session
 
+from logger import LOG
+
 
 def is_admin_interface():
     return os.getenv("ADMIN", "false") == "true"
 
 
+def has_admin_role():
+    is_admin_role = False
+    group = session.get("group", None)
+    if group:
+        is_admin_role = "admin-" in group["value"]
+    return is_admin_role
+
+
+def current_group_name():
+    default_group_name = "group-not-available"
+    group_name = default_group_name
+    group = session.get("group", None)
+    if group:
+        group_name = group.get("value", default_group_name)
+    return group_name
+
+
 def is_development():
     return os.getenv("FLASK_ENV", "production") == "development"
 
 
 def admin_interface(flask_route):
     @wraps(flask_route)
     def decorated_function(*args, **kwargs):
-
         if not is_admin_interface():
-            raise Exception("ADMIN not set when trying /admin")
+            session["error_message"] = "The requested route is not available"
+            LOG.error({"action": "access denied", "reason": "ADMIN env var missing"})
+            return redirect("/403")
+        if not has_admin_role():
+            session["error_message"] = "User not authorised to access this route"
+            LOG.error(
+                {
+                    "action": "access denied",
+                    "reason": "User not authorised",
+                    "username": session.get("user", None),
+                }
+            )
+            return redirect("/403")
         return flask_route(*args, **kwargs)
 
     return decorated_function
 
 
+def end_user_interface(flask_route):
+    @wraps(flask_route)
+    def decorated_function(*args, **kwargs):
+        if has_admin_role():
+            # raise Exception("User not authorised to access this route")
+            session["error_message"] = "User not authorised to access this route"
+            return redirect("/403")
+        return flask_route(*args, **kwargs)
+
+    return decorated_function
+
+
+def requires_group_matching(role_like):
```

Comment:
For the same reason as above (and with the same caveat as you probably knowing more than I) feel uneasy about using regexes for these matches, there are only six user groups so an explicit list is (currently) always going to be short and will avoid any unintended consequences

Reply:
Agreed.